### PR TITLE
working fix for Set whileLoadingPage prop

### PR DESCRIPTION
**Pull request by daniel-escoto**
Fix for https://github.com/redwoodjs/redwood/issues/6763.

**Problem:**
1. Set components have no `whileLoadingPage` prop, even though [RedwoodJS documentation](https://redwoodjs.com/docs/router#page-loaders--pageloadingcontext) mentions them.
2. All child routes of Set components with a valid `whileLoadingProp` input fail to render said prop when page is loading.

**Current Status**:
`WhileLoadingProp` is now functional, but some `router.test.tsx` tests are failing; currently looking into why this is and how to fix it. Failed tests can be found here: https://cloud.nx.app/runs/SFh6eO3T8J/task/@redwoodjs%2Frouter:test

### CODE REVIEW

File: packages/router/src/Set.tsx [hunk 1]
```diff
@@ -45,10 +47,16 @@ export function Set<WrapperProps>(props: SetProps<WrapperProps>) {
...
     ...rest
   } = props
   const routerState = useRouterState()
-  const { loading, isAuthenticated, hasRole } = routerState.useAuth()
+  const {
+    loading: authLoading,
+    isAuthenticated,
+    hasRole,
+  } = routerState.useAuth()
+  const { loading: pageLoading } = usePageLoadingContext()
```
Comment (daniel-escoto) [inline packages/router/src/Set.tsx:59]: This line causes tests to fail... still unsure why.

Comment (daniel-escoto) [reply packages/router/src/Set.tsx:59]: @Tobbe Still a bit stuck on this, would love to know if you have any feedback, or have seen [errors like these](https://cloud.nx.app/runs/SFh6eO3T8J/task/@redwoodjs%2Frouter:test) before.

**Comment by Tobbe** [comment]
I've been looking at this code for a while now.

It clearly works when I try it in an actual project. But something's not working the same way in our tests.
`loading` is never set to `true` in the page loading context. Haven't figured out exactly why yet.

This is from `active-route-loader.tsx`
```
      // Update the context if importing the page is taking longer
      // than `delay`.
      // Consumers of the context can show a loading indicator
      // to signal to the user that something is happening.
      loadingTimeout.current = setTimeout(() => {
        unstable_batchedUpdates(() => {
          setLoadingState((loadingState) => ({
            ...loadingState,
            [path]: {
              page: whileLoadingPage || ArlWhileLoadingNullPage,
              specName: '',
              state: 'SHOW_LOADING',
              location,
            },
          }))
          setRenderedChildren(children)
          setRenderedPath(path)
        })
      }, delay)
```

The timeout callback is never executed. I tried using `0` instead of `delay`. Didn't help. Also tried using `act()` in the test case in various ways/places. No luck.

**Comment by Tobbe** [comment]
I had a hunch, and it's looking promising so far 🙂 I'll run some more tests and then share some code if all test pass

**Comment by Tobbe** [comment]
Tests looked good, so I took the liberty to push to this PR. You can see my changes in this commit https://github.com/redwoodjs/redwood/pull/6912/commits/5a451447839cdb561976e27cbe12228f744bbc86

**Comment by daniel-escoto** [comment]
Thanks @Tobbe!

**Pull request merged**